src/js/libgeo/GeoOps.js

```diff
@@ -170,6 +170,29 @@ geoOps.Vertical.updatePosition = function(el) {
 };
 
 
+geoOps.LineByFixedAngle = {};
+geoOps.LineByFixedAngle.kind = "L";
+geoOps.LineByFixedAngle.initialize = function(el) {
+    var a = CSNumber._helper.input(el.angle);
+    var c = CSNumber.cos(a);
+    var s = CSNumber.sin(a);
+    el.rot = List.turnIntoCSList([
+        List.turnIntoCSList([s, c]),
+        List.turnIntoCSList([CSNumber.neg(c), s])
+    ]);
+};
```

Comment:
Storing the rotation as a matrix makes sense, I think. However, the matrix is not the one I expected (which would be `[[c, -s], [s, c]]`) so it took me a moment to understand why this works. The reason it does is that you start with the coordinates of a line, apply the transformation and treat the result as a point. This reinterpretation adds another 90° so to speak. A comment might help future readers of this code.

Seeing how you do a lot of coordinate-wise manipulation in `updatePosition`, I'd say make this a 3×3 matrix to start with:

```
      ⎡ s  c  0⎤
dir = ⎢-c  s  0⎥ ∙ l.homog
      ⎣ 0  0  0⎦
```

This will make the code below simpler by two lines.